ddl/index_change_test.go

```diff
@@ -124,10 +126,19 @@ func (s *testIndexChangeSuite) TestIndexChange(c *C) {
 			if err != nil {
 				checkErr = errors.Trace(err)
 			}
+			if job.State == model.JobStateSynced {
+				addIndexDone = true
+			}
 		}
 	}
 	d.SetHook(tc)
 	testCreateIndex(c, ctx, d, s.dbInfo, originTable.Meta(), false, "c2", "c2")
+	for i := 0; i <= 10; i ++ {
+		if addIndexDone {
+			break
+		}
+		time.Sleep(50 * time.Millisecond)
```

Comment:
Don't get the point here, when `testCreateIndex()` is done, isn't `addIndexDone` has been set to `true` already?

Reply:
No, The jobState may not be JobStateSynced

Reply:
Please add comments on why we need this check.